### PR TITLE
KAA-1124 Update documentation structure

### PR DESCRIPTION
**Pull request by Dubland**
/cc @rasendubi @forGGe

### CODE REVIEW

File: doc/Customization-guide/Kaa-Server/Development-environment-setup/index.md [hunk 1]
```diff
@@ -182,7 +181,7 @@ It is also important to provide your class with the @KaaPluginConfig annotation.
 **NOTE**  
 A log appender descriptor is optional if you are going to configure your log appenders using only REST API.
 
-The following code example illustrates the implementation of a log appender descriptor. 
+The following code example illustrates the implementation of a log appender descriptor.
```
Comment (rasendubi) [inline doc/Customization-guide/Kaa-Server/Development-environment-setup/index.md]: Remove trailing space.

File: doc/Customization-guide/Kaa-Server/Development-environment-setup/index.md [hunk 2]
```diff
@@ -258,9 +257,9 @@ To implement a custom transport, you need to complete the following steps.
 3.  Implement [Transport](https://github.com/kaaproject/kaa/tree/v0.9.0/server/common/transport-shared/src/main/java/org/kaaproject/kaa/server/transport/Transport.java).
 4.  Provision the transport in a Bootstrap server and/or an Operations server.
 
-We recommend that you use one of the existing [transport implementations](https://github.com/kaaproject/kaa/tree/v0.9.0/server/transports) as a reference and also review the [transports design reference999](999).
+We recommend that you use one of the existing [transport implementations](https://github.com/kaaproject/kaa/tree/v0.9.0/server/transports) as a reference and also review the [transports design reference999](999).
 
-Please note that once a new transport is implemented, you will most likely need to [implement a corresponding transport channel](#creating-custom-transport-channel) for one or multiple endpoint SDK platforms. 
+Please note that once a new transport is implemented, you will most likely need to [implement a corresponding transport channel](#creating-custom-transport-channel) for one or multiple endpoint SDK platforms.
```
Comment (balldir) [inline doc/Customization-guide/Kaa-Server/Development-environment-setup/index.md]: Is it a trailing whitespace ?

File: doc/Customization-guide/Kaa-Server/Development-environment-setup/index.md [hunk 3]
```diff
@@ -567,29 +566,29 @@ Notice that in java sdk, after adding new data channel multiplexer and demultipl
 
 To send a request to the server and get a response please follow the steps described below.
 
-## Step 1 - Get an instance of KaaDataMultiplexer 
+## Step 1 - Get an instance of KaaDataMultiplexer
```
Comment (balldir) [inline doc/Customization-guide/Kaa-Server/Development-environment-setup/index.md]: Is it a trailing whitespace ?

File: doc/Customization-guide/Kaa-Server/Development-environment-setup/index.md [hunk 4]
```diff
@@ -891,4 +890,4 @@ To provision your user verifier, do the following:
 
 1.  Place the user verifier descriptor and configuration classes into the Admin UI class path.
 2.  Place the user verifier implementation classes into the Operations Server class path.
-3.  Use [Admin UI999](https://docs.kaaproject.org/display/KAA/Administration+UI+guide#AdministrationUIguide-Userverifiers) or [REST API999](https://docs.kaaproject.org/display/KAA/Admin+REST+API#AdminRESTAPI-Create/edituserverifier) to create/update/delete your user verifier instances.
+3.  Use [Admin UI999](https://docs.kaaproject.org/display/KAA/Administration+UI+guide#AdministrationUIguide-Userverifiers) or [REST API999](https://docs.kaaproject.org/display/KAA/Admin+REST+API#AdminRESTAPI-Create/edituserverifier) to create/update/delete your user verifier instances.
```
Comment (balldir) [inline doc/Customization-guide/Kaa-Server/Development-environment-setup/index.md:893]: Please create a separate ticket to update this links.

Comment (rasendubi) [inline doc/Customization-guide/Kaa-Server/Development-environment-setup/index.md]: It was discussed this change goes under KAA-1124

Comment (Dubland) [inline doc/Customization-guide/Kaa-Server/Development-environment-setup/index.md:893]: This guide is not yet ready and is a boilerplate. Links will be fixed in the jira for the guide.

**Pull request merged**